build/buildBuildImages.sh

```diff
@@ -171,120 +87,190 @@ function createImageNameWithReleaseTag() {
 	# Retag build image with build number tags
 	if [ "$AGENT_BUILD" == "true" ]
 	then
-		local uniqueImageName="$imageNameToBeTaggedUniquely-$BUILD_DEFINITIONNAME.$RELEASE_TAG_NAME"
+		IFS=':' read -ra IMAGE_NAME <<< "$imageNameToBeTaggedUniquely"
+		local repo="${IMAGE_NAME[0]}"
+		local tag="$BUILD_DEFINITIONNAME.$RELEASE_TAG_NAME"
+		if [ ${#IMAGE_NAME[@]} -eq 2 ]; then
+			local uniqueImageName="$imageNameToBeTaggedUniquely-$tag"
+		else
+			local uniqueImageName="$repo:$tag"
+		fi
 
 		echo
-		echo "Retagging image '$imageNameToBeTaggedUniquely' with ACR related tags..."
+		echo "Retagging image '$imageNameToBeTaggedUniquely' as '$uniqueImageName'..."
 		docker tag "$imageNameToBeTaggedUniquely" "$uniqueImageName"
 
 		# Write image list to artifacts file
 		echo "$uniqueImageName" >> $ACR_BUILD_IMAGES_ARTIFACTS_FILE
 	fi
 }
 
+function buildGitHubRunnersBaseImage() {
+	echo
+	echo "----Building the image which uses GitHub runners' buildpackdeps-stretch specific digest----------"
+	docker build -t githubrunners-buildpackdeps-stretch \
+		-f "$BUILD_IMAGES_GITHUB_RUNNERS_BUILDPACKDEPS_STRETCH_DOCKERFILE" \
+		.
+}
+
+function buildTemporaryFilesImage() {
+	buildGitHubRunnersBaseImage
+	
+	# Create the following image so that it's contents can be copied to the rest of the images below
+	echo
+	echo "-------------Creating temporary files image-------------------"
+	docker build -t tempfiles \
+		-f "$BUILD_IMAGES_TEMPORARY_FILES_DOCKERFILE" \
+		.
+}
+
 function buildBuildScriptGeneratorImage() {
+	buildTemporaryFilesImage
+
 	# Create the following image so that it's contents can be copied to the rest of the images below
 	echo
 	echo "-------------Creating build script generator image-------------------"
 	docker build -t buildscriptgenerator \
 		--build-arg AGENTBUILD=$BUILD_SIGNED \
-		$buildMetadataArgs \
+		--build-arg GIT_COMMIT=$GIT_COMMIT \
```

Comment:
Need not worry about these args as the image being built here is an intermediate image only.